### PR TITLE
Dra snapshot stage

### PR DESCRIPTION
**Pull request by sharbuz**
## What is the problem this PR solves?

The partial migration of the Jenkins pipeline to the Buildkite pipeline. 

## How does this PR solve the problem?

Migration of the "DRA Snapshot" step of the "packaging.groovy" pipeline.

## How to test this PR locally

It doesn't be checked locally, that's why it's a partial "step by step" migration

## Checklist

The Buildkite pipeline will be triggered by this PR and we will see the result

## Related issues

Related to the issue: https://github.com/elastic/fleet-server/issues/2517

**Pipeline steps:**
- [x] Pipeline init
- [x] Filter build - Skipped because this filter is required by Jenkins auto-scan and we already have all additional conditions for starting the pipeline
- [x] Package
- [x] DRA Snapshot
- [ ] DRA Release Staging

### CODE REVIEW

File: .buildkite/scripts/package.sh
Comment (sharbuz) [reply .buildkite/scripts/package.sh:34]: @mrodm , almost all these things were fixed in the [PR](https://github.com/elastic/fleet-server/pull/2651), I'm waiting for your final decision (approval) according to PR-2651, then I'll merge those changes to the main branch and to this branch as well.

File: .buildkite/pipeline.package.mbp.yml
Comment (mrodm) [inline .buildkite/pipeline.package.mbp.yml:43]: mmm when this step runs, post-checkout hook is not executed yet.
How does this variable get the value ?

Maybe those functions needs to be moved as part of the script `dra_snapshot.sh` ?

File: .buildkite/scripts/release-manager.sh
Comment (sharbuz) [reply .buildkite/scripts/release-manager.sh:24]: It's a good question, I forgot to ask it you and Alex. I need to understand what contains in the file, perhaps something important. @mrodm , @alexsapran, who can help with getting this file from Jenkins, `ci-systems` channel??

File: .buildkite/hooks/post-checkout
Comment (sharbuz) [reply .buildkite/hooks/post-checkout:57]: I placed this part in the post-checkout hook because I need to define the variable `IS_FILE_EXISTS_IN_REPO` before the pipeline initialization (uploading stage). The `rda-snapshot` stage depends on the value of this variable: https://github.com/elastic/fleet-server/blob/4b95d6b1c3e091da204e86df7c8b797a10144495/.buildkite/pipeline.package.mbp.yml#L43

File: .buildkite/hooks/post-checkout
Comment (sharbuz) [reply .buildkite/hooks/post-checkout:78]: Answered here:
https://github.com/elastic/fleet-server/pull/2702#discussion_r1239715942

File: .buildkite/hooks/post-checkout
Comment (sharbuz) [reply .buildkite/hooks/post-checkout:76]: It was renamed because, in fact, it's checking a file in the repository, not a branch.

File: .buildkite/pipeline.package.mbp.yml
Comment (sharbuz) [reply .buildkite/pipeline.package.mbp.yml:43]: If I'm not wrong, the post-checkout step is runningg before uploading the pipeline to the agent.
I tested it when I changed the file name value of this variable stayed `false`: https://github.com/elastic/fleet-server/pull/2702/commits/be236a42626a9b769847e55034ec7180dbd0e788

File: .buildkite/pipeline.package.mbp.yml
Comment (sharbuz) [reply .buildkite/pipeline.package.mbp.yml:5]: I don't know why you are seeing that...
https://github.com/elastic/fleet-server/blob/4b95d6b1c3e091da204e86df7c8b797a10144495/.buildkite/pipeline.package.mbp.yml#L1-L7

File: .buildkite/hooks/post-checkout
Comment (sharbuz) [reply .buildkite/hooks/post-checkout:37]: I thought about it, and I'm not sure that it's a better way to add the source file to the hook `common.sh` for only one func.
If it's OK, no problem, I will do that.

File: .buildkite/scripts/common.sh
Comment (sharbuz) [reply .buildkite/scripts/common.sh]: fixed

File: .buildkite/scripts/release-manager.sh
Comment (mrodm) [reply .buildkite/scripts/release-manager.sh:24]: Don't know where that file would come from... it could be checked with them, if that file would exist in the agents.

File: .buildkite/hooks/pre-command
Comment (mrodm) [inline .buildkite/hooks/pre-command:46]: As there is no permission here for now, should be set some fake values so the build can be run ? 


```suggestion
    # DRA_CREDS=$(retry 5 vault kv get -field=data -format=json "${DRA_SECRET_PATH}")
    # export VAULT_ADDR=$(echo ${DRA_CREDS} | jq -r '.vault_addr')
    # export VAULT_ROLE_ID=$(echo ${DRA_CREDS} | jq -r '.vault_role_id')
    # export VAULT_SECRET_ID=$(echo ${DRA_CREDS} | jq -r '.vault_secret_id')
    # TODO just set dummy values until vault secrets can be retrieved.
    DRA_CREDS="fake_path"
    export VAULT_ADDR="vault_address_dummy"
    export VAULT_ROLE_ID="vault_role_id_dummy"
    export VAULT_SECRET_ID="vault_secret_id_dummy"
```

File: .buildkite/hooks/post-checkout
Comment (mrodm) [reply .buildkite/hooks/post-checkout:37]: I would say that `common.sh` file could be sourced here, AFAIK it should just contain some functions that should be helpers. Or, is there any action run when that file is sourced ? @sharbuz

File: .buildkite/hooks/post-checkout
Comment (mrodm) [reply .buildkite/hooks/post-checkout:78]: Would this also work if it is moved to the `pre-command` hook ?
It looks like this check fits better to that other hook, than the post-checkout.
WDYT?

File: .buildkite/hooks/post-checkout
Comment (sharbuz) [reply .buildkite/hooks/post-checkout:57]: I tested it and we can repeat the test together.

File: .buildkite/scripts/release-manager.sh
Comment (sharbuz) [reply .buildkite/scripts/release-manager.sh:24]: ok, I'll ask in the `ci-systems` channel.

File: .buildkite/pipeline.package.mbp.yml
Comment (sharbuz) [reply .buildkite/pipeline.package.mbp.yml]: fixed

File: .buildkite/hooks/post-checkout
Comment (mrodm) [inline .buildkite/hooks/post-checkout:52]: If `branch` is not main, `branchName variable is not created.

What value should have that `branchName` variable if `branch` is not `main`?

File: .buildkite/hooks/post-checkout
Comment (mrodm) [reply .buildkite/hooks/post-checkout]: Thanks for testing it again @sharbuz . Nice to see that some variables can be calculated in the post-checkout hook and then used in the own step definition

File: .buildkite/hooks/post-checkout
Comment (mrodm) [reply .buildkite/hooks/post-checkout]: I see that it has been changed to `FILE_EXISTS_IN_REPO` :+1:

File: .buildkite/hooks/post-checkout
Comment (sharbuz) [reply .buildkite/hooks/post-checkout:53]: fixed

**Review by mrodm — APPROVED** [review]
LGTM

**Pull request merged**
